src/components/data-management/AddMetadataButton.jsx

```diff
@@ -61,7 +66,7 @@ const AddMetadataButton = ({ samplesTableRef }) => {
       console.log(e);
     }
     setCellLevelUploading(false);
-    setCellLevelUploadVisible(false);
+    return file;
```

Comment:
I see you are returning `file` in here. Where is this file used later in?

Reply:
The fileObject generated when compressing the file will be reused in case the Retry button is pressed. No need for doing everything again. The same happens for the sample file upload